src/triangle.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2024 the Kurbo Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Triangle shape
+use crate::{PathEl, Point, Rect, Shape, Vec2};
+
+use core::cmp::*;
+use core::f64::consts::FRAC_PI_4;
+use core::ops::{Add, Sub};
+
+#[cfg(not(feature = "std"))]
+use crate::common::FloatFuncs;
+
+/// Triangle
+//     A
+//     *
+//    / \
+//   /   \
+//  *-----*
+//  B     C
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Triangle {
+    /// vertex a
+    pub a: Point,
+    /// vertex b
+    pub b: Point,
+    /// vertex c
+    pub c: Point,
+}
+
+impl Triangle {
+    /// The empty [`Triangle`] at the origin
+    pub const ZERO: Self = Self::from_coords((0., 0.), (0., 0.), (0., 0.));
+
+    /// empty [`Triangle`] at (1.0, 1.0)
+    pub const ONE: Self = Self::from_coords((1.0, 1.0), (1.0, 1.0), (1.0, 1.0));
+
+    /// equilateral [`Triangle`]
+    pub const EQUILATERAL: Self = Self::from_coords(
+        (
+            1.0 / 2.0,
+            1.732050807568877293527446341505872367_f64 / 2.0, /* (sqrt 3)/2 */
+        ),
+        (0.0, 0.0),
+        (1.0, 0.0),
+    );
+
+    /// A new [`Triangle`] from three vertices ([`Points`])
+    #[inline]
+    pub fn new(a: impl Into<Point>, b: impl Into<Point>, c: impl Into<Point>) -> Self {
+        Self {
+            a: a.into(),
+            b: b.into(),
+            c: c.into(),
+        }
+    }
+
+    /// A new [`Triangle`] from three float vertex coordinates
+    #[inline]
+    pub const fn from_coords(a: (f64, f64), b: (f64, f64), c: (f64, f64)) -> Self {
```

Comment:
Ive used it to keep the constants `ZERO` and `EQUILATERAL`. 

Maybe there is a way to use the non constant `new()` since we can assume the arguments are already of `Point`. If there isnt another way then should i remove them or maybe i can make `from_coords()` private?

Reply:
The easiest way is probably to just inline this function at the definitions. I.e. `const ZERO: Triangle = Triangle {a: Point::new(...), ...}`.

I do agree that there should be some consistency here, but I'm happy just using the already-exposed "fieldwise" constructor as the `const` constructor for now to unblock progress.

Reply:
Yeah im happy to do that. should we quickly do it now or wait for @derekdreery's input?

Reply:
Hmm I hadn't realised that this function exists for `const`. In that case I think it's good to have 2 functions,  but perhaps put a note in the docs explaining that this function works like `new`, but can be used in `const` contexts.